src/test/java/com/answerdigital/answerking/controller/CategoryControllerTest.java

```diff
@@ -225,12 +235,52 @@ void updateCategoryWithInvalidCategoryRequestDescTest() throws Exception {
         final var categoryRequest = "{\"name\": \"random name\",\"description\": \"random description #\"}";
 
         final String error = Objects.requireNonNull(mvc.perform(put("/categories/{categoryId}", 112L)
-            .content(categoryRequest)
-            .contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE))
-            .andExpect(status().isBadRequest())
-            .andExpect(content().contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE))
-            .andReturn().getResolvedException()).getMessage();
+                        .content(categoryRequest)
+                        .contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE))
+                .andReturn().getResolvedException()).getMessage();
 
         assertTrue(error.contains("Category description can only contain letters, numbers, spaces and !?-.,' punctuation"));
     }
+
+    @Test
+    void getCategoryByIdReturnsOkStatusIfExist() throws Exception {
+        //given
+        final CategoryResponse categoryResponse = categoryResponseTestBuilder.withDefaultValues().build();
+        when(categoryService.findByIdResponse(1L)).thenReturn(categoryResponse);
+        //when
+        final ResultActions actualPerformResult = mvc.perform(get("/categories/{categoryId}", 1L)).andExpect(status().isOk());
+        final ObjectMapper mapper = new ObjectMapper();
+        //then
+        assertEquals(categoryResponse.getId(), mapper.readTree(actualPerformResult.andReturn()
+                .getResponse().getContentAsString()).get("id").asLong());
+    }
+
+    @Test
+    void getAllCategoriesReturnListOfCategoryObjects() throws Exception {
+        //given
+        final CategoryResponse categoryResponse = categoryResponseTestBuilder.withDefaultValues().build();
+
+        given(categoryService.findAll()).willReturn(Set.of(categoryResponse));
+
+        //when
+        final RequestBuilder request = MockMvcRequestBuilders.get("/categories");
+
+        final MockHttpServletResponse response = mvc.perform(request).andReturn().getResponse();
+
+        //then
+        assertEquals(HttpStatus.OK.value(), response.getStatus());
+        assertFalse(response.getContentAsString().isEmpty());
+        final ObjectMapper mapper = new ObjectMapper();
```

Comment:
I think it would be good to move 
final ObjectMapper mapper = new ObjectMapper();
final JsonNode jsonNodeResponse = mapper.readTree(response.getContentAsString());

a bit more above to separate it from the assertion section. There are more such places.